indigo/bulk_creator.py

```diff
@@ -47,6 +48,19 @@ class RowValidationFormBase(forms.Form):
     repealed_by = forms.CharField(required=False)
     taxonomy = forms.CharField(required=False)
 
+    def __init__(self, country, locality, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['country'].choices = [(country.code, country.name)]
+        self.fields['locality'].choices = [(locality.code, locality.name)] \
+            if locality else []
+        self.fields['doctype'].choices = self.get_doctypes_for_country(country.code)
+        self.fields['subtype'].choices = [(s.abbreviation, s.name) for s in SUBTYPES]
+
+    def get_doctypes_for_country(self, country_code):
+        return [([d[1], d[0]]) for d in
```

Comment:
Is this correctly nested? It's going to produce, eg: `[(['x', 'y'], ['a', 'b'])]` -- I think you have too many brackets.

Reply:
Oh, no, it actually works because the parentheses don't land up making a tuple. But they can be removed completely.

```suggestion
        return [[d[1], d[0]] for d in
```